crates/bdk/tests/wallet.rs

```diff
@@ -88,6 +87,62 @@ fn test_get_funded_wallet_balance() {
     assert_eq!(wallet.get_balance().confirmed, 50000);
 }
 
+#[test]
+fn test_get_funded_wallet_sent_and_received() {
+    let (wallet, _) = get_funded_wallet(get_test_wpkh());
+    assert_eq!(wallet.get_balance().confirmed, 50000);
+    let mut tx_amounts: Vec<(Txid, (u64, u64))> = wallet
+        .transactions()
+        .map(|ct| (ct.node.txid, wallet.sent_and_received(ct.node.tx)))
+        .collect();
+    tx_amounts.sort_by(|a1, a2| a1.0.cmp(&a2.0));
+
+    assert_eq!(tx_amounts.len(), 2);
+    assert_matches!(tx_amounts.get(0), Some((_, (76_000, 50_000))))
+}
+
+#[test]
+fn test_get_funded_wallet_tx_fees() {
+    let (wallet, _) = get_funded_wallet(get_test_wpkh());
+    assert_eq!(wallet.get_balance().confirmed, 50000);
+    let mut tx_fee_amounts: Vec<(Txid, Result<u64, CalculateFeeError>)> = wallet
+        .transactions()
+        .map(|ct| {
+            let fee = wallet.calculate_fee(ct.node.tx);
+            (ct.node.txid, fee)
+        })
+        .collect();
+    tx_fee_amounts.sort_by(|a1, a2| a1.0.cmp(&a2.0));
+
+    assert_eq!(tx_fee_amounts.len(), 2);
+    assert_matches!(
+        tx_fee_amounts.get(1),
+        Some((_, Err(CalculateFeeError::MissingTxOut(_))))
+    );
+    assert_matches!(tx_fee_amounts.get(0), Some((_, Ok(1000))))
```

Comment:
How did we know this fees here? Would it be possible to send a tx from `bitcoind` and have `bitcoind` give us the fee/feerate and we can compare the result with what we get from BDK?

Reply:
Since we're just calling rust-bitcoin's calculations here I don't think we need to re-validate their code. But I can document the calculation for anyone who wants to see where the final number comes from.